base/docs/basedocs.jl

```diff
@@ -693,4 +693,23 @@ run(pipeline(`cat test.txt`, DevNull))
 """
 DevNull
 
+"""
+    Function
+
+Abstract type for which all functions are instances. This includes operators, functions
```

Comment:
Yup, "Abstract type of all functions" seems pretty clear to me too. @ararslan perhaps the examples can be changed to make it clear that functions have their own type and are always a subtype of `Function`. I think we can rely on the rest of the documentation to make it clear that operators, lambdas and functions are all just functions in Julia.